### PR TITLE
feat: add "as" prop to logo

### PR DESCRIPTION
**Pull request by ndv99**
## Done
- Added "as" prop to Logo (similar to Link)
- Created types.ts and moved AsProp here

## QA steps

- [ ] Ensure the navigation renders correctly

### CODE REVIEW

**Review by petermakowski — APPROVED** [review]
Solid work.

File: src/lib/sections/Navigation/Logo/Logo.tsx [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 
-export const Logo = ({ children }: NavigationLogoProps) => {
+export const Logo = <
+C extends ElementType = "a",
+T extends ComponentProps<C> = ComponentProps<C>,
+>({ as, children, className, ...props }: AsProp<C> & Omit<T, "as">) => {
+  const Component = as || "a";
    return (
-    <a className="p-panel__logo" href="/">
+    <Component className={classNames("p-panel__logo", className)} {...props}>
```
Comment (petermakowski) [inline src/lib/sections/Navigation/Logo/Logo.tsx:17]: One suggestion I have is we could create a HOC e.g. `withAsProp` to reduce duplication (and reuse in Link component).

**Review by Jay-Topher — APPROVED** [review]
LGTM

File: src/lib/sections/Navigation/Logo/Logo.tsx (hunk 1 above)
Comment (ndv99) [reply src/lib/sections/Navigation/Logo/Logo.tsx:17]: Did some investigating around this and I think it needs some more time/work. I'll merge this in its current state and create a jira ticket.

**Pull request merged**